docs/gguf.md

```diff
@@ -0,0 +1,382 @@
+
+If a particular community key is widely used, it may be promoted to a standardized key.
+
+### General
+
+- **`general.architecture: string`**: describes what architecture this model implements. All lowercase ASCII, with only `[a-z0-9]+` characters allowed. Known values include:
+  - `llama`
+  - `mpt`
+  - `gptneox`
+  - `gptj`
+  - `gpt2`
+  - `bloom`
+  - `falcon`
+  - `rwkv`
+- **`general.quantization_version: u32`**: version of quantization scheme
+- `general.file_type: string`: type of the majority of the tensors in the file. This shouldn't have any semantic meaning and should be purely informational, hence the use of `string`.
+- `general.license: string`: SPDX license of the model
+- `general.description: string`: free-form description of the model including anything that isn't covered by the other fields
+- `general.source.url: string`: URL to the source of the model. Can be a GitHub repo, a paper, etc.
+- `general.source.huggingface.repository: string`: Hugging Face model repository that this model is either hosted on or based on
+
+### LLM
+
+In the following, `[llm]` is used to fill in for the name of a specific LLM architecture. They will be used in each architecture's section.
+
+- `[llm].context_length: u32`: size of the maximum supported context
+- `[llm].hidden_size: u32`: embedding layer size
+- `[llm].num_layers: u32`: number of layers
+- `[llm].num_ff: u32`: The length of the feedforward layer.
+- `[llm].use_parallel_residual: bool`: whether or not the parallel residual logic should be used
+- `[llm].max_seq_len: u32`: Maximum sequence length
+- `[llm].attention.num_heads: u32`: number of attention heads
+- `[llm].attention.alibi_bias_max: f32`: The maximum bias to use for ALiBI
+- `[llm].attention.clip_kqv: f32`: Value (`C`) to clamp the values of the `Q`, `K`, and `V` tensors between (`[-C, C]`).
+- `[llm].rope.num_dims: u32`: The number of rotary dimensions for RoPE.
+
+#### Models
+
+The following sections describe the metadata for each model architecture. Each key specified _must_ be present.
+
+##### LLaMA
+
+- `llama.context_length`
+- `llama.hidden_size`
+- `llama.num_layers`
+- `llama.num_ff`
+- `llama.rope.num_dims`
+- `llama.attention.num_heads`
+
+##### MPT
+
+- `mpt.max_seq_len`
+- `mpt.hidden_size`
+- `mpt.num_layers`
+- `mpt.attention.num_heads`
+- `mpt.attention.alibi_bias_max`
+- `mpt.attention.clip_kqv`
+
+##### GPT-NeoX
+
+- `gptneox.context_length`
+- `gptneox.hidden_size`
+- `gptneox.num_layers`
+- `gptneox.use_parallel_residual`
+- `gptneox.rope.num_dims`
+- `gptneox.attention.num_heads`
+
+##### GPT-J
+
+- `gptj.context_length`
+- `gptj.hidden_size`
+- `gptj.num_layers`
+- `gptj.rope.num_dims`
+- `gptj.attention.num_heads`
+
+##### GPT-2
+
+- `gpt2.context_length`
+- `gpt2.hidden_size`
+- `gpt2.num_layers`
+- `gpt2.attention.num_heads`
+
+##### BLOOM
+
+- `bloom.context_length`
+- `bloom.hidden_size`
+- `bloom.num_layers`
+- `bloom.num_ff`
+- `bloom.attention.num_heads`
+
+##### Falcon
+
+**TODO**.
```

Comment:
Falcon needs:

falcon.max_seq_len
falcon.hidden_size
falcon.num_layers
falcon.attention.num_heads
falcon.attention.num_heads_kv (only in 40b model)